### PR TITLE
[percep -> nav] Use point cloud to better retrieve AR tag distance and bearing

### PR DESCRIPTION
**Pull request by qhdwight**
Previously we were using just the pixel information + camera FOV to calculate bearing. Distance was calculated using the depth, so it was actually inaccurate and would be worse the less aligned you were with the gate. Now the actual point cloud information from the ZED is being used to calculate both of these better.

The intent of these fixes were to improve the accuracy of nav's estimation of gate positions since it uses both of these values. They should still be filtered on nav's end.

### CODE REVIEW

**Comment by qhdwight** [comment]
As per usual view with whitespace off: https://github.com/umrover/mrover-workspace/pull/1072/files?diff=split&w=1

**Review by niwhsa9 — APPROVED** [review]
[https://media.giphy.com/media/gLu90OMjz4j3hAkJk2/giphy.gif](url)

**Comment by qhdwight** [comment]
```c++
int y_pixel = tags.y_pixels.at(i);
int x_pixel = tags.x_pixels.at(i);
float x, y, z;
{
    auto xyz = xyz_img.at<Vec4f>(y_pixel, x_pixel);
    x = xyz[0];
    y = xyz[1];
    z = xyz[2];
}
auto raw_depth = depth_img.at<float>(y_pixel, x_pixel);
if (!isnan(raw_depth)) {
    arTags[i].distance = sqrt(x * x + y * y + z * z) * static_cast<float>(MM_PER_M);
}
// +z is forward, +x is right, all relative to camera
arTags[i].bearing = atan2(x, z);
arTags[i].id = tags.ids.at(i);
tags.buf_counts[i] = 0;
```

this is the important part of this pr

File: jetson/percep/main.cpp [hunk 1]
```diff
@@ -7,59 +7,59 @@
 using namespace cv;
 using namespace std;
```
Comment (cgiger00) [inline jetson/percep/main.cpp:8]: would you mind getting rid of this? bad practice

Comment (qhdwight) [reply jetson/percep/main.cpp:8]: yeah it was in the original code but I can remove the `using namespace`s

Comment (cgiger00) [reply jetson/percep/main.cpp:8]: thx

**Comment by cgiger00** [comment]
do you want to hold on off merging this before this PR https://github.com/umrover/mrover-workspace/pull/962 is approved?

**Comment by qhdwight** [comment]
> do you want to hold on off merging this before this PR #962 is approved?

Yes that would probably be good

**Pull request merged**